auth/auth.go

```diff
@@ -7,34 +7,42 @@ import (
 
 // Auth providers authentication and authorization
 type Auth interface {
-	// Generate a new auth token
-	Generate(string) (*Token, error)
-	// Revoke an authorization token
-	Revoke(*Token) error
-	// Grant access to a resource
-	Grant(*Token, *Service) error
-	// Verify a token can access a resource
-	Verify(*Token, *Service) error
+	// Init the auth package
+	Init(opts ...Option) error
+	// Generate a new auth ServiceAccount
+	Generate(*ServiceAccount) (*ServiceAccount, error)
+	// Revoke an authorization ServiceAccount
+	Revoke(string) error
+	// Validate a service account token
+	Validate(string) (*ServiceAccount, error)
 }
 
-// Service is some thing to provide access to
-type Service struct {
-	// Name of the resource
-	Name string
-	// Endpoint is the specific endpoint
-	Endpoint string
+// Resource is an entity such as a user or service
+type Resource struct {
+	// Id of the resource
+	Id string
+	// Type of resource, e.g. Service
+	Type string
 }
 
-// Token providers by an auth provider
-type Token struct {
-	// Unique token id
-	Id string `json: "id"`
-	// Time of token creation
+// Role a service account has
+type Role struct {
+	Name     string
+	Resource *Resource
+}
+
+// ServiceAccount providers by an auth provider
+type ServiceAccount struct {
+	// The parent of the service account, e.g. a user
+	Parent *Resource `json: "parent"`
```

Comment:
Would maybe just call this Id? as in Account Id?

Reply:
We aren't storing the parent anywhere yet so what would the ID be?

Reply:
Actually, we could do type (user) and name (asim@micro.mu). What would we use the ID for?

Reply:
Every account should have an Id, whether thats defined as email or service name or whatever. And then maybe a "Root" or something in metadata to define an association to a different account.

Reply:
Agree, i'll action this now